qt/widgets/instrumentview/inc/MantidQtWidgets/InstrumentView/Shape2D.h

```diff
@@ -301,6 +302,60 @@ class Shape2DRing : public Shape2D {
   double m_yWidth;
 };
 
+/**
+ * A sector: area defined by an inner and an outer arcs of circle, interrupted
+ * at angles startAngle and endAngle.
+ *
+ * The constructor takes a center QPoint property, 2 angles, and 2 widths for
+ * the two circles.
+ */
+class Shape2DSector : public Shape2D {
+public:
+  Shape2DSector(double inner_radius, double outer_radius, double start_angle,
+                double end_angle, const QPointF &center);
+  Shape2DSector(const Shape2DSector &sector);
+  Shape2D *clone() const override { return new Shape2DSector(*this); }
+  bool selectAt(const QPointF &p) const override;
+  bool contains(const QPointF &p) const override;
+  // double properties
+  QStringList getDoubleNames() const override;
+  double getDouble(const QString &prop) const override;
+  void setDouble(const QString &prop, double value) override;
+  // QPointF properties
+  QStringList getPointNames() const override { return QStringList("center"); }
+  QPointF getPoint(const QString &prop) const override;
+  void setPoint(const QString &prop, const QPointF &value) override;
+
+  /// Because setting the new bounding box makes no sense, this is just overrode
+  /// and does nothing at all
+  virtual void setBoundingRect(const RectF &rect) override { UNUSED_ARG(rect); }
+
+  /// Load state for the shape from a project file
+  static Shape2D *loadFromProject(const std::string &lines);
+  /// Save state for the shape to a project file
+  virtual std::string saveToProject() const override;
+  std::string type() const override { return "sector"; }
+
+protected:
+  void drawShape(QPainter &painter) const override;
+  void addToPath(QPainterPath & /*path*/) const override {}
+  void refit() override;
+  void resetBoundingRect() override;
+  void computeScaling(QPointF BBoxCorner, QPointF BBoxOpposedCorner,
+                      QPointF bRectCorner, int vertexIndex);
+  size_t getShapeNControlPoints() const override { return 4; }
+  QPointF getShapeControlPoint(size_t i) const override;
+  void setShapeControlPoint(size_t i, const QPointF &pos) override;
+  double m_innerRadius;
+  double m_outerRadius;
+  double m_startAngle;
+  double m_endAngle;
+  QPointF m_center;
+};
+
+QRectF findArcBoundingBox(double startAngle, double endAngle);
```

Comment:
Why are these two out of the class? What is the namespace? If these are needed only for sector put them as private members in sector class.